index.html

```diff
@@ -441,6 +441,171 @@ <h2><dfn>GamepadEvent</dfn> Interface</h2>
+          that same window.
+        </p>
+
+    </section>
+
+    <section>
+
+        <h3 id="event-gamepadbuttondown">The <dfn class="event">gamepadbuttondown</dfn> event</h3>
+
+        <p>
+          User agents implementing this specification must provide a new DOM
+          event, named <a><code>gamepadbuttondown</code></a>. The
+          corresponding event MUST be of type
+          <a href="#gamepadevent-interface"><code>GamepadEvent</code></a>
+          and MUST fire on the <code>window</code> object. Registration for
+          and firing of the <a><code>gamepadbuttondown</code></a> event MUST
+          follow the usual behavior of DOM4 Events. [[!DOM4]]
+        </p>
+
+        <p>
+          A <a>user agent</a> MUST dispatch this event type to indicate the
+          user has pressed down an individual button on a gamepad.
+        </p>
+
+        <p>
+          If a gamepad was already connected when the page was loaded but the
+          <a>user agent</a> has not yet dispatched a
+          <a href="#event-gamepadconnected"><code>gamepadconnected</code></a>
+          to a window, the
+          <a href="#event-gamepadconnected"><code>gamepadconnected</code></a>
+          event MUST be dispatched before dispatching the
+          <a><code>gamepadbuttondown</code></a> event to that same window.
+        </p>
+
+        <p>
+          The <a><code>gamepadbuttondown</code></a> event type must be
+          dispatched before the
+          <a href="#event-gamepadbuttonup"><code>gamepadbuttonup</code></a>
+          event associated with the same key.
+        </p>
+
+    </section>
+
+    <section>
+
+        <h3 id="event-gamepadbuttonup">The <dfn class="event">gamepadbuttonup</dfn> event</h3>
+
+        <p>
+          User agents implementing this specification must provide a new DOM
+          event, named <a><code>gamepadbuttonup</code></a>. The corresponding
+          event MUST be of type
+          <a href="#gamepadbuttonevent-interface"><code>GamepadButtonEvent</code></a>
+          and MUST fire on the <code>window</code> object. Registration for
+          and firing of the <a><code>gamepadbuttonup</code></a> event MUST
+          follow the usual behavior of DOM4 Events. [[!DOM4]]
+        </p>
+
+        <p>
+          A <a>user agent</a> MUST dispatch this event type to indicate the
+          user has released an individual button on a gamepad.
+        </p>
+
+        <p>
+          The <a><code>gamepadbuttonup</code></a> event type must be
+          dispatched after the
+          <a href="#event-gamepadbuttondown"><code>gamepadbuttondown</code></a>
+          event associated with the same key.
+        </p>
+
+    </section>
+
+    <section>
+
+        <h3>Other events</h3>
+
+        <p>
+
+          <i>More discussion is needed on whether to include events for axis
+          changes (e.g., <code>gamepadaxismove</code> for whenever the value of
```

Comment:
Please also consider adding a gamepadbuttonmove event in order to better support gamepads that expose buttons with analog values.

Reply:
good call, thanks!